std/cs/_std/sys/thread/Semaphore.hx

```diff
@@ -0,0 +1,36 @@
+package sys.thread;
+
+@:coreApi
+class Semaphore {
+	final native:SemaphoreSlim;
+
+	public function new(value:Int):Void {
+		this.native = new SemaphoreSlim(value);
+	}
+
+	public function acquire():Void {
+		native.Wait();
+	}
+
+	public function tryAcquire(?timeout:Float):Bool {
+		return native.Wait(timeout == null ? 0 : Std.int(timeout * 1000));
+	}
+
+	public function release():Void {
+		native.Release();
+	}
+}
+
+// doesn't seem to be in the dlls shipped with hxcs?
+
+@:native("System.Threading.SemaphoreSlim")
+private extern class SemaphoreSlim implements cs.system.IDisposable {
```

Comment:
SemaphoreSlim was added in .NET 4.0. Our Haxe targets 2.0 by default.

Reply:
It uses `System.Threading.Semaphore` now.